Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -4978,11 +4979,13 @@ - (void)collectionView:(UICollectionView *)collectionView
 - (void)contactsPickerDidCancel:(ContactsPicker *)contactsPicker
 {
     DDLogDebug(@"%@ in %s", self.logTag, __PRETTY_FUNCTION__);
+    [self dismissViewControllerAnimated:YES completion:nil];
```

Comment:
> // TODO - there are some hard coded assumptions in this VC that assume we are *pushed* onto a
> // navigation controller.

Here we see the inverse of what I was rambling about in the `AddContactShareToExisting` class. 

The ContactsPicker was similarly hardcoding assumptions about it's dismissal. Previously it was only ever used in a modal context, so all these dismisses were hardcoded into the ContactsPicker. Since *here* we were pushing the picker view, the existing dismissal logic didn't work.

On average, I think things work out better when the same view that pushes/presents a VC is responsible for popping/dismissing it - typically as part of a delegate method (even if it means some duplication).

Reply:
Agree - for VCs with delegates.